### PR TITLE
feat(repair/newsletter): added newsletter component

### PR DESCRIPTION
**Pull request by mserwo**

### CODE REVIEW

File: src/components/footer/join-newsletter/join-newsletter.tsx [hunk 1]
```diff
@@ -3,6 +3,29 @@ import { SingleInputField } from "../single-input-field/single-input-field";
...
+      if (response.ok) {
+        setEmailSent(true);
+        console.log("Email sent successfully!");
+      } else {
+        console.error("Failed to send email");
+      }
+    } catch (error) {
+      console.error("Error sending email:", error);
+    }
+  };
```
Comment (GabrLit) [inline src/components/footer/join-newsletter/join-newsletter.tsx]: Remove console.logs, they should not be present in Pull Request, 
Secondly move this whole functionality to another file, example: `api/newsletter.ts` const postNewsletter = ()=>{}
Third, catching the error means something went wrong so we should notify the user, if we are setting setEmailSent we should also show Error to the user `Coś poszło nie tak` or anything like this

File: src/server/app.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+const cors = require("cors");
+const app = express();
+const port = 3000;
+
+app.use(cors());
+app.use(express.json());
+
+app.post("/marcin", (request, response) => {
+  const { email } = request.body;
+  console.log("Odebrałem email:", email);
```
Comment (GabrLit) [inline src/server/app.js]: no console logs is PR here too

File: src/server/app.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  response.send("Email został odebrany!");
```
Comment (GabrLit) [inline src/server/app.js]: we dont really have to send ("Email został odebrany!") we are not reading this string on frontend. the practice is to send back what has been added to a database, we are not adding to database but we can pretend.

response.send(200).(email)

File: src/server/app.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+});
+
+app.listen(port, () => {
+  console.log(`Example app listening on port ${port}`);
```
Comment (GabrLit) [inline src/server/app.js:15]: This is the one console log that can stay

**Review by GabrLit — APPROVED** [review]
Looks good, some minor changes required

File: src/server/app.js (hunk 3 above)
Comment (mserwo) [reply src/server/app.js]: response.status(200).send(email); 
as GPT says: in send() we can put only sending content. Status ok, we should put in status()

File: src/server/app.js (hunk 2 above)
Comment (mserwo) [reply src/server/app.js]: ok :(

File: src/components/footer/join-newsletter/join-newsletter.tsx (hunk 1 above)
Comment (mserwo) [reply src/components/footer/join-newsletter/join-newsletter.tsx]: Secondly: i have setter in this function :( I'll get everything mixed up
Third: i have console.error here. Its not ok?

File: package.json [hunk 5]
```diff
@@ -13,12 +13,14 @@
...
   "dependencies": {
+    "classnames": "^2.5.1",
+    "cors": "^2.8.5",
     "localforage": "^1.10.0",
     "match-sorter": "^6.3.1",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-router-dom": "^6.10.0",
-    "sort-by": "^1.2.0",
+    "sort-by": "^0.0.2",
```
Comment (GabrLit) [inline package.json:23]: why do we need this package?

File: src/components/footer/join-newsletter/join-newsletter.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        body: JSON.stringify({ email: value }),
+      });
+
+      if (response.ok) {
+        setEmailSent(true);
+      } else {
+        setServerError(true);
+      }
+    } catch (error) {
+      console.error("Error sending email:", error);
```
Comment (GabrLit) [inline src/components/footer/join-newsletter/join-newsletter.tsx]: we are still displaying console log here, why

Comment (GabrLit) [reply src/components/footer/join-newsletter/join-newsletter.tsx]: if we catch error we should display to the user Something went wrong, just the same if the response is not okay

Comment (GabrLit) [reply src/components/footer/join-newsletter/join-newsletter.tsx]: so instead of console.error , setServerError(true)

**Pull request merged**